lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -208,19 +214,45 @@ private void sendAuditEvent(
             AuditEventTypes auditEventType,
             CoiCheckType coiCheckType,
             Boolean coiCheckSuccess,
-            String userId,
-            String ipvSessionId,
-            String govukSigninJourneyId,
-            String ipAddress)
-            throws SqsException {
-        var auditEventUser =
-                new AuditEventUser(userId, ipvSessionId, govukSigninJourneyId, ipAddress);
+            AuditEventUser auditEventUser,
+            List<VerifiableCredential> oldVcs,
+            List<VerifiableCredential> sessionsVcs,
+            String deviceInformation)
+            throws SqsException, HttpResponseExceptionWithErrorBody, CredentialParseException {
+
+        var restrictedData =
+                auditEventType == AuditEventTypes.IPV_CONTINUITY_OF_IDENTITY_CHECK_END
+                        ? getRestrictedCheckCoiAuditData(oldVcs, sessionsVcs, deviceInformation)
+                        : new AuditRestrictedDeviceInformation(deviceInformation);
 
         auditService.sendAuditEvent(
                 new AuditEvent(
                         auditEventType,
                         configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
                         auditEventUser,
-                        new AuditExtensionCoiCheck(coiCheckType, coiCheckSuccess)));
+                        new AuditExtensionCoiCheck(coiCheckType, coiCheckSuccess),
+                        restrictedData));
+    }
+
+    private AuditRestrictedCheckCoi getRestrictedCheckCoiAuditData(
+            List<VerifiableCredential> oldVcs,
+            List<VerifiableCredential> sessionVcs,
+            String deviceInformation)
+            throws HttpResponseExceptionWithErrorBody, CredentialParseException {
+        var oldIdentityClaim = userIdentityService.findIdentityClaim(oldVcs);
+        var sessionIdentityClaim = userIdentityService.findIdentityClaim(sessionVcs);
+
+        if (oldIdentityClaim.isEmpty() || sessionIdentityClaim.isEmpty()) {
+            LOGGER.error(LogHelper.buildLogMessage("Failed to get identity claim."));
+            return new AuditRestrictedCheckCoi(
+                    null, null, null, null, new DeviceInformation(deviceInformation));
+        }
```

Comment:
If you want to combine these two blocks, you could also do:
```
return new AuditRestrictedCheckCoi(
        oldIdentityClaim.map(IdentityClaim::getName).orElse(null),
        sessionIdentityClaim.map(IdentityClaim::getName).orElse(null),
        oldIdentityClaim.map(IdentityClaim::getBirthDate).orElse(null),
        sessionIdentityClaim.map(IdentityClaim::getBirthDate).orElse(null),
        new DeviceInformation(deviceInformation));
```
Sadly Java makes this sort of thing quite awkward - in TS it would be a much clearer `oldIdentityClaim?.getName()`